### ISSUE TITLE
Nested RSX expressions

### ISSUE THREAD

**Issue by chichid** [stoically/syn-rsx#19]
Would it be possible (or is it planned) to support nested RSX expression within Blocks, without having to specify rsx! for each sub expression.

For instance can this: 
`
quote! {
<View>
    {
         children_array.map(|child| {
              let my_child = rsx! { <Test /> }
              some other code...
              my_child
         })
    }
</View>
`

Be simplified to this (without the nested RSX)
quote! {
<View>
    {
         children_array.map(|child| {
              let my_child = <Test />
              some other code...
              my_child
         })
    }
</View>
`

**Comment by stoically** [comment]
I've asked something similar upstream (https://github.com/dtolnay/syn/issues/793) and the conclusion was no, unfortunately. I'd be interested in a solution if someone happens to stumble over one.

**Comment by stoically** [comment]
I don't think that works. The crucial part is https://github.com/dtolnay/syn/issues/793#issuecomment-630224255.

```rust
let uwu = <div></div>
```

simply isn't valid rust code, which means it won't compile and won't be passed to a proc macro.

**Comment by chichid** [comment]
yeah that's what I'm asking above. For sure attribute-like macros won't support it.

**Comment by stoically** [comment]
I see. Yeah, I guess having the ability to hook into certain points in `syn`s parsing might work without adding JSX parsing itself to syn.

**Comment by chichid** [comment]
Closing this bug for now as it answers my question, I may reopen it if I contribute something to syn and makes it through. But syn_rsx is definitely the right step into bringing true JSX ergonomics to the rust world.
